src/mount.ts

```diff
@@ -516,6 +516,24 @@ export function mount(
     app.mixin(mixin)
   }
 
+  // mixin for initial data of script setup composition components
+  if (options?.data) {
+    const providedData = options.data()
+    const mixin = defineComponent({
+      beforeCreate() {
+        if (hasSetupState(this)) {
+          for (const [k, v] of Object.entries(
+            providedData as { [key: string]: any }
+          )) {
+            this.$.setupState[k] = v
+          }
+        }
+      }
+    })
+
+    app.mixin(mixin)
```

Comment:
Wouldn't this be called for every component? Not only the component which has been passed to `mount`

Reply:
you are right, as i added this as a mixin it will be called for every component in this App.
I will have a look if there is another convenient way to set the data fields.